HW3/Exceptions/UsersNotFoundException.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.IO;
+
+namespace HW3.Exceptions
+{
+    internal class UsersNotFoundException : FileNotFoundException
+    {
+        public UsersNotFoundException(string? message, string? filename) : base(message, filename)
```

Comment:
Maybe a full message is not needed, when you can form it out of a missing filename? You could at least make it optional and there doesn't seem to be much value in always providing a different one.